### PR TITLE
refactor: use VS Code authentication to login Azure

### PR DESCRIPTION
**Pull request by xiaolang124**
Use VS Code authentication to login to Azure.

1.The login process will involve two sequential modal dialogs to proceed.
image

image

2.Teams toolkit will not support sign out Azure in tree view, users need to sign out Azure account in account menu.
image

### CODE REVIEW

File: packages/vscode-extension/package.json [hunk 1]
```diff
@@ -1406,10 +1411,11 @@
     "react-copy-to-clipboard": "^5.1.0",
     "react-syntax-highlighter": "^15.5.0",
     "validator": "^13.7.0",
-    "vscode-tas-client": "^0.1.75"
+    "vscode-tas-client": "^0.1.75",
+    "@azure/arm-resources-subscriptions": "^2.1.0",
```
Comment (swatDong) [inline packages/vscode-extension/package.json:1415]: Keep the dependencies in alphabetical order for better maintenance

File: packages/vscode-extension/src/commonlib/vscodeAzureSubscriptionProvider.ts [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    return subscriptions;
+  }
+}
+
+export async function getSessionFromVSCode(
+  scopes?: string | string[],
+  tenantId?: string,
+  options?: vscode.AuthenticationGetSessionOptions
+): Promise<vscode.AuthenticationSession | undefined> {
+  return await vscode.authentication.getSession(Microsoft, getScopes(scopes, tenantId), options);
```
Comment (swatDong) [inline packages/vscode-extension/src/commonlib/vscodeAzureSubscriptionProvider.ts:132]: What's the minimal vsc version supporting this new auth flow? Do we need to update the `engines -> vscode` in ext's *package.json*?

File: packages/vscode-extension/src/commonlib/vscodeAzureSubscriptionProvider.ts [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+  Custom = "custom",
+}
+
+/**
+ * Gets the configured Azure environment.
+ *
+ * @returns The configured Azure environment from the settings in the built-in authentication provider extension
+ */
+export function getConfiguredAzureEnv(): azureEnv.Environment & { isCustomCloud: boolean } {
+  const authProviderConfig = vscode.workspace.getConfiguration(CustomCloudConfigurationSection);
```
Comment (swatDong) [inline packages/vscode-extension/src/commonlib/vscodeAzureSubscriptionProvider.ts:191]: I believe some toolkit features do not support non-public azure. Do we need to block logging into non-public azure?

File: packages/vscode-extension/package.nls.json [hunk 4]
```diff
@@ -24,7 +24,7 @@
     "teamstoolkit.appStudioLogin.loginCancel": "Sign-in canceled. Teams Toolkit needs a Microsoft 365 account with custom app upload permission. If you're a Visual Studio subscriber, create a developer sandbox with the Microsoft 365 Developer Program (https://developer.microsoft.com/en-us/microsoft-365/dev-program).",
     "teamstoolkit.appStudioLogin.message": "Teams Toolkit needs a Microsoft 365 account with custom app upload permission. If you're a Visual Studio subscriber, create a developer sandbox with the Microsoft 365 Developer Program.",
     "teamstoolkit.azureLogin.failToFindSubscription": "We couldn't find a subscription.",
-    "teamstoolkit.azureLogin.message": "The Teams Toolkit requires an Azure account and subscription to deploy the Azure resources for your project. You won't be charged until you confirm.",
+    "teamstoolkit.azureLogin.message": "The Teams Toolkit will use Microsoft authentication to sign in Azure account and subscription to deploy the Azure resources for your project. You will not be charged without further confirmation.",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:27]: I'd propose to change the second line to the existing one i.e. You won't be charged until you confirm.
Logic: We use contractions i.e. won't, can't, you've, etc.as per our style guide. And, the existing line is easier to perceive, has a personalized tone, and clear. The revised line doesn't convey clearly - without whose further confirmation?

File: packages/vscode-extension/package.nls.json [hunk 5]
```diff
@@ -59,6 +59,7 @@
     "teamstoolkit.commands.accountsLink.title": "Learn More About Accounts",
     "teamstoolkit.commands.addAppOwner.title": "Add Microsoft 365 Teams App (with Microsoft Entra App) Owners",
     "teamstoolkit.commands.azureAccountSettings.title": "Azure Portal",
+    "teamstoolkit.commands.azureAccount.signOutHelp": "Azure account Sign Out has been moved to the Accounts section on the bottom left panel. To sign out Azure, hover on your Azure account email address and click Sign Out.",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:62]: Proposed: Azure account Sign Out is moved to the Accounts section on the bottom left panel. To sign out of Azure, hover on your Azure account email and click Sign Out.
Logic: Simplified it by eliminating passive voice and making it little crisper to improve readability.

**Review by supkasar — APPROVED** [review]

File: packages/vscode-extension/src/commonlib/vscodeAzureSubscriptionProvider.ts [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+export class VSCodeAzureSubscriptionProvider {
+  private async getSubscriptionClient(
+    tenantId?: string,
+    scopes?: string[]
+  ): Promise<{
+    client: SubscriptionClient;
+    credential: TokenCredential;
+    authentication: AzureAuthentication;
+  }> {
+    const armSubs = await import("@azure/arm-resources-subscriptions");
```
Comment (tecton) [inline packages/vscode-extension/src/commonlib/vscodeAzureSubscriptionProvider.ts:24]: What is the effect to use dynamic import here?

**Review by tecton — APPROVED** [review]

**Pull request merged**